roadmap/implementers-guide/src/runtime/validity.md

```diff
@@ -17,8 +17,37 @@ We account for these requirements by having the validity module handle two kinds
+
+We liberate availability cores when their candidate becomes available of course, but one approval assignment criteria continues associating each candidate with the core number it occupied when it became available. 
+
+Assignment proceeds in loosely timed rounds called `DelayTranche`s roughly 12 times faster than block production, in which validators send assignment notices until all candidates have enough checkers assigned.  Assignment tracks when approval votes arrive too and assigns more checkers if some checkers run late.
+
+Approval checks provide more security than backing checks, so polkadot becomes more efficient when validators perform more approval checks per backing check.  If validators run 4 approval checks for every backing check, and run almost one backing check per relay chain block, then validators actually check almsot 6 blocks per relay chain block.
+
+We should therefore reward approval checkers correctly because approval checks should actually represent our single largest workload.  It follows that both assignment notices and approval votes should be tracked on-chain.  
+
+We might track the assignments and approvals together as pairs in a simple rewards system.  There are however two reasons to track assignments and approvals separately on-chain:
+
+- An approval that arrives slowly prompts assigning extra checkers.  Yet, we consider a block valid if the earlier checker completes their work, even if the extra checkers never quite finish, which complicates rewarding these extra checkers.  We could support more nuanced rewards for extra checkers if assignments are placed on-chain earlier.  Assignment delay tranches progress 12ish times faster than the relay chain, but no shows could still be witness by the relay chain because the no show delay takes longer than a relay chain slot. 
+
+- We know off-chain when the approval process completes based upon all gossiped assignment notices, not just the approving ones.  We need not-yet-approved assignment notices to appear on-chain if the chain should know about the validity of recently approved blocks.  Integration with grandpa then requires only that grandpa messages identify some future relay chain block that approves the block for which they vote. 
+
+Aside from inherents for assignment notices and approval votes, we provide an "Approved" inherent that triggers the on-chain logic in a relay chain block `R1` to rerun the assignment and approval tracker logic for its ancestor `R0`, which then declares `R0` approved.  We could integrate with GRANDPA by gossiping messates that list the descendent `R1`, but then map this into the approved ancestor `R0` for GRANDPA itself.
```

Comment:
"inherent" is probably not the right terminology if you're referring to an automated action taken by the chain automatically

Reply:
> triggers the on-chain logic in a relay chain block `R1` to rerun the assignment and approval tracker logic for its ancestor `R0`

This might have a lot of hidden complexity due to state pruning

Reply:
```suggestion
Aside from inherents for assignment notices and approval votes, we provide an "Approved" inherent that triggers the on-chain logic in a relay chain block `R1` to rerun the assignment and approval tracker logic for its ancestor `R0`, which then declares `R0` approved.  We could integrate with GRANDPA by gossiping messages that list the descendent `R1`, but then map this into the approved ancestor `R0` for GRANDPA itself.
```